### YamlMime:JavaType
uid: "com.azure.resourcemanager.applicationinsights.models.ErrorDefinition"
fullName: "com.azure.resourcemanager.applicationinsights.models.ErrorDefinition"
name: "ErrorDefinition"
nameWithType: "ErrorDefinition"
summary: "Error response."
inheritances:
- "<xref href=\"java.lang.Object?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.management.exception.ManagementError?displayProperty=fullName\" data-throw-if-not-resolved=\"False\" />"
inheritedMembers:
- "com.azure.core.management.exception.ManagementError.getAdditionalInfo()"
- "com.azure.core.management.exception.ManagementError.getCode()"
- "com.azure.core.management.exception.ManagementError.getDetails()"
- "com.azure.core.management.exception.ManagementError.getMessage()"
- "com.azure.core.management.exception.ManagementError.getTarget()"
- "com.azure.core.management.exception.ManagementError.toString()"
- "java.lang.Object.clone()"
- "java.lang.Object.equals(java.lang.Object)"
- "java.lang.Object.finalize()"
- "java.lang.Object.getClass()"
- "java.lang.Object.hashCode()"
- "java.lang.Object.notify()"
- "java.lang.Object.notifyAll()"
- "java.lang.Object.wait()"
- "java.lang.Object.wait(long)"
- "java.lang.Object.wait(long,int)"
syntax: "public final class ErrorDefinition extends ManagementError"
constructors:
- uid: "com.azure.resourcemanager.applicationinsights.models.ErrorDefinition.ErrorDefinition()"
  fullName: "com.azure.resourcemanager.applicationinsights.models.ErrorDefinition.ErrorDefinition()"
  name: "ErrorDefinition()"
  nameWithType: "ErrorDefinition.ErrorDefinition()"
  syntax: "public ErrorDefinition()"
methods:
- uid: "com.azure.resourcemanager.applicationinsights.models.ErrorDefinition.getInnererror()"
  fullName: "com.azure.resourcemanager.applicationinsights.models.ErrorDefinition.getInnererror()"
  name: "getInnererror()"
  nameWithType: "ErrorDefinition.getInnererror()"
  summary: "Get the innererror property: Internal error details."
  syntax: "public Object getInnererror()"
  desc: "Get the innererror property: Internal error details."
  returns:
    description: "the innererror value."
    type: "<xref href=\"java.lang.Object?alt=java.lang.Object&text=Object\" data-throw-if-not-resolved=\"False\" />"
- uid: "com.azure.resourcemanager.applicationinsights.models.ErrorDefinition.validate()"
  fullName: "com.azure.resourcemanager.applicationinsights.models.ErrorDefinition.validate()"
  name: "validate()"
  nameWithType: "ErrorDefinition.validate()"
  summary: "Validates the instance."
  syntax: "public void validate()"
  desc: "Validates the instance."
type: "class"
desc: "Error response."
metadata: {}
package: "com.azure.resourcemanager.applicationinsights.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-applicationinsights:1.0.0-beta.5
